osu.Game/Overlays/Wiki/Markdown/WikiMarkdownImageBlock.cs

```diff
@@ -45,5 +42,50 @@ private void load()
                 }),
             };
         }
+
+        private class BlockMarkdownImage : WikiMarkdownImage
```

Comment:
Is there a reason for not doing this in the o!f base implementation? I'm not sure that images overflowing the bounds of the container are ever going to be useful.

Reply:
Because there's two type of image.
- Inline image, it should be `AutoSizeAxes.Both` (current implementation of `MarkdownImage` in o!f).
- Block image, it should be `AutoSizeAxes.Y` and `RelativeSizeAxes.X`.

Reply:
But why couldn't the handling of everything be moved to o!f?

That is, I suggest moving this to o!f and exposing a `MarkdownContainer.CreateImageBlock()` virtual method:
https://github.com/ppy/osu/blob/5edaf8235d5a4d300676ec52ed17edd02747f9b7/osu.Game/Overlays/Wiki/Markdown/WikiMarkdownContainer.cs#L34-L43

I think most/all implementations of markdown are going to require the same scaling logic?

Reply:
Yeah, that would do.

I just don't wanna change the implementation in o!f-side when code this.

Reply:
Hmm okay, well it can be moved at any point then.